### ISSUE TITLE
When ignoring MD033no-inline-html, MD034no-bare-urls always triggers in urls inside html tags

### ISSUE THREAD

**Issue by WurmD** [markdownlint/markdownlint#435]
## Description
The MD034no-bare-urls rule does not work inside html tags, it always triggers even if the url is not bare

### Environment
Conda environment, python 3.8.13
**MDL Version**: 0.11.0

### Expected Behavior

With MD033no-inline-html rule ignored, the following should pass
```markdown
<details>
  <summary>Terraform:</summary>

1. [Install Terraform](<https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli>)
1. ...

</details>
```

### Actual Behavior
`README.md:4: MD034 Bare URL used`

## Replication Case

`mkdir MDtest`
`cd MDtest`
create `README.md` with

```markdown
<details>
  <summary>Terraform:</summary>

1. [Install Terraform](<https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli>)
1. ...

</details>
```

```bash
cat > .pre-commit-config.yaml << EOF
---
default_language_version:
  python: python3.9

repos:
- repo: https://github.com/markdownlint/markdownlint.git
  rev: v0.11.0
  hooks:
  - id: markdownlint
    args: ["-r", "~MD033"]   # MD033: Allow "Inline HTML"
EOF

git init .
git add .
pre-commit run --all
```
